xpk.py

```diff
@@ -230,365 +237,416 @@
 """
 
 cluster_configmap_yaml = """kind: ConfigMap
-apiVersion: v1 
+apiVersion: v1
 metadata:
-  name: {args.cluster}-resources-configmap 
+  name: {args.cluster}-resources-configmap
 data:
   {data}
 """
 
+# @dataclass
+# class AcceleratorType(Enum):
+#     TPU = 1
+#     GPU = 2
+#     # CPU = 3,
+
+AcceleratorType = {
+  'TPU': 1,
+  'GPU': 2
+}
+
+@dataclass
+class AcceleratorCharacteristics:
+  resource_type: str
+  accelerator_label: int
+  machine_label: str
+
+AcceleratorTypeToAcceleratorCharacteristics = {
+     # TPU
+      AcceleratorType['TPU']: AcceleratorCharacteristics(
+      'google.com/tpu', 'cloud.google.com/gke-tpu-accelerator', 'cloud.google.com/gke-tpu-topology'
+      ),
+     # GPU
+      AcceleratorType['GPU']: AcceleratorCharacteristics(
+      'nvidia.com/gpu', 'cloud.google.com/gke-accelerator', 'cloud.google.com/gce-machine-type'
+      )
+}
+
+
 @dataclass
 class SystemCharacteristics:
   topology: str
   vms_per_slice: int
   gke_accelerator: str
   gce_machine_type: str
   chips_per_vm: int
+  accelerator_type: AcceleratorType
+  device_type: str
 
 ################### Subcommand Helper Functions #############
 """ !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 IF YOU MODIFY THE BELOW UserFacingNameToSystemCharacteristics MAP YOU SHOULD ALSO ADD CORRESPONDING
 MODIFICATIONS TO UserFacingNameToSystemCharacteristics IN MaxText/accelerator_to_spec_map.py !!!!! """
 # vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
 UserFacingNameToSystemCharacteristics = {
+    # GPU system charcteristics
+    # A100-40gb
```

Comment:
nit: can you change # A100-40gb to # A100-40gb-$CHIPS
This will help hint to what the last postfix represents

Reply:
Same for # H100-80gb